server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -765,7 +771,10 @@ private void construct(
             MetadataCreateDataStreamService.class,
             new MetadataCreateDataStreamService(threadPool, clusterService, metadataCreateIndexService)
         );
-        modules.bindToInstance(MetadataDataStreamsService.class, new MetadataDataStreamsService(clusterService, indicesService));
+        modules.bindToInstance(
```

Comment:
I think it's worth separating out this and DataStreamGlobalRetentionResolver creation into a separate method (similar to `createScriptService` or `createUpdateService`)

Reply:
So, I was looking into it and it's not clear what I am trying to achieve with these methods. The options I see:

- Bundle together the following into `initialiseDataStreamServices(...)`:
```
        modules.bindToInstance(
                MetadataCreateDataStreamService.class,
                new MetadataCreateDataStreamService(threadPool, clusterService, metadataCreateIndexService)
        );
        modules.bindToInstance(
                MetadataDataStreamsService.class,
                new MetadataDataStreamsService(clusterService, indicesService, dataStreamGlobalRetentionResolver)
        );
        ```
- Extract also the `DataStreamGlobalRetentionResolver` in a separate method on its own.
```
DataStreamGlobalRetentionResolver dataStreamGlobalRetentionResolver = new DataStreamGlobalRetentionResolver(
            DataStreamFactoryRetention.load(pluginsService, clusterService.getClusterSettings())
        );
        modules.bindToInstance(DataStreamGlobalRetentionResolver.class, dataStreamGlobalRetentionResolver);
``` 
Do the above align with what you expected?